scarlet/display.py

```diff
@@ -321,74 +355,76 @@ def show_sources(
             observation is not None
         ), "Provide matched observation to show observed frame"
 
-    panels = 1 + sum((show_observed, show_rendered, show_sed))
+    panels = sum((show_model, show_observed, show_rendered, show_sed))
     if figsize is None:
         figsize = (3 * panels, 3 * len(list(sources)))
     fig, ax = plt.subplots(len(list(sources)), panels, figsize=figsize)
     for k, src in enumerate(sources):
-
         if hasattr(src, "center"):
-            center = np.array(src.center)
-            # center in src bbox coordinates
-            if src.bbox is not None:
-                center_ = center - np.array(src.bbox.origin[1:])
-            else:
-                center_ = center
-            # center in observed coordinates
-            center__ = center - np.array(src.frame.origin[1:])
+            center = np.array(src.center)[::-1]
         else:
             center = None
 
         panel = 0
-        frame_ = src.frame
-        src.set_frame(src.bbox)
+
+        model = src.get_model()
         if isinstance(src, ComponentTree):
-            model = 0
             seds = []
             for component in src:
                 model_ = component.get_model()
                 seds.append(model_.sum(axis=(1, 2)))
-                model += model_
         else:
-            model = src.get_model()
             seds = [model.sum(axis=(1, 2))]
-        src.set_frame(frame_)
-        ax[k][panel].imshow(img_to_rgb(model, norm=norm, channel_map=channel_map, mask=model.sum(axis=0)==0))
-        ax[k][panel].set_title("Model Source {}".format(k))
-        if center is not None:
-            ax[k][panel].plot(*center_[::-1], "wx", mew=1, ms=10)
 
-        if show_rendered:
+        if use_mask:
+            mask = model.sum(axis=0) == 0
```

Comment:
no! models are not masked. It's visual clutter and conceptually improper to define the mask this way (what if we *want* the model to be zero over some range). I suggest that we set masking consistently in all panels, i.e. we set the observation mask in all panels if `use_mask is True`.

Reply:
Actually, it should be an argument `mask=None` like in `show_scene` above.

Reply:
This is something that I specifically need, so I included it here as an option (but no longer the default). When I'm investigating how a particular change is affecting the models I want to know how big the full extent of the model is, if it runs up against the edge of the bounding box, if it is weird or lopsided, etc. Otherwise I have to write my own method in an extension module to do the exact same thing except pass this one option, so I'd rather just keep it here. So I differentiate it here because this `mask` is not the same as the mask-plane `mask` that marks where there is no data to constrain the model.

Reply:
The full extent of the model is its bbox. Checking for zero value is a check that assumes a particular constraint, i.e. very special purpose. I suggest that you change in favor of specifying the mask explicitly and then provide `mask = model.sum(axis=0) == 0` if you need that for something. This way the code is much more generally useful.

Reply:
Actually, I still don't understand what you would be using the zero-mask for. I get it for PSFs, but not for regular source models.

Reply:
I agree that this mask should be passed in the parameters as opposed to enforcing a the mask for a particular configuration of a model (even if it is the most commonly used model), so I'll update that. And I agree that for the PSF convolved model we want to see which pixels were masked in the observation, which is why we use that mask consistently in `show_scene`.

But when I look at a particular source I want to know how the underlying model behaved. Note that we don't use sparsity, so any zero pixels in an objects `2^n` x `2^n` box really have no flux, whereas pixels with even a small amount of flux (`10^-large number`) will be black and not masked. So it lets me see where the model wants to put power and helps me understand how/why it is doing that (for example because there was a neighbor there). This effect gets blurred out with the PSF, plus the PSF convolution makes this particular mask go away. So it is something that I use for a specific purpose, but use often enough that I don't want to maintain a separate function on my own (I already have to do that with source initialization).

Reply:
It might have some diagnostic value (although I find pixels around zero rarely diagnostically useful), but certainly not enough to justified it being baked into code as it is.

Reply:
Even the current version pushed today?

Reply:
It's fine now